specification/theory_operations.adoc

```diff
@@ -1,2 +1,430 @@
 [[theory_operations]]
+
 == Theory of Operations
+
+The CoVE-IO specification extends the CoVE host and guest SBI extensions to
+allow TVMs to establish trust with TEE-IO devices, and then use and interact
+with those devices. Untrusted supervisor domain components are responsible for
+assigning TEE-IO devices to TVMs, and also for supporting the TVM acceptance or
+rejection of the assigned devices. The TSM, on the other hand, establishes and
+maintains the secure physical and logical links between TVMs and their assigned
+devices.
+
+The following sections describe the functionality of the TSM-provided CoVE-IO
+extensions to support trusted I/O on CoVE-enabled platforms.
+
+=== Platform Initialization
+
+==== IOMMU Registration and Setup
+
+The TSM relies on the availability of at least one IOMMU instance exclusvely
+associated with the TSM supervisor domain. Those IOMMUs allow the TSM to enforce
+the integrity of address translations and protection from DMA into confidential
+memory, as well as  interrupts originating from assigned TDIs. The host
+supervisor domain may assign one or more IOMMU instances to the TSM supervisor
+domain, after which, only the TSM can access and program the assigned IOMMU
+instances.
+
+IOMMUs assigned to the TSM security domain may generate MSIs in order to signal
+the TSM about command completions, transaction faults or device page requests.
+Those MSIs target system physical memory, which is owned by the host security
+domain manager, e.g. the host VMM. As a consequence, it is the host security
+domain's responsibility to reserve the MSI addresses and then rely on the TSM to
+program the IOMMUs with those reserved addresses. This IOMMU registration
+process is driven by the untrusted domain manager for all IOMMUs that
+participate in TEE-IO and operates as described in the following steps:
+
+1. The TSM is loaded into a supervisor domain and provisioned with a CoVE-IO
+   manifest. It is recommended that the TSM is measured by the platform
+   root-of-trust for subsequent attestation.
```

Comment:
Question: Why TSM is measured by platform RoT, but not silicon RoT (CPU)?

Reply:
perhaps we should say SoC RoT?

Reply:
I think it depends on who is Root-of-Trust for measurement (RTM), and who loads TSM.
If TSM is loaded by host CPU, the SoC RoT makes more sense.
Unless TSM is really loaded by platform RoT, then platform RoT needs to measure it. And the platform RoT also needs to guarantee that the one CPU executing on is the same as the one loaded and measured by Platform RoT.

If we support both use case, then we can generalize it by saying: TSM is measured by Root-of-Trust for measurement (RTM). Maybe...